### PR TITLE
Changes for DLC support

### PR DESCRIPTION
**Pull request by vmednonogov**
Ticket: 14368

This PR works together with PR https://github.com/carbonated-dev/o3de-gruber/pull/1080

## What does this PR do?

Issue (1):
On iOS the first level ("launch" level) was launched before the engine CrySystem is initialized and before the system UI canvas is loaded. It prevented to show confirmation DLC downloading message box on iOS and prevented to show DLC download progress bar on iOS (they are initialized on the ESYSTEM_EVENT_LEVEL_LOAD_END event).

Changes (1): 
For iOS DLC support - move the first level launch from SpawnableLevelSystem constructor into time after CrySystem is initialized and RPI is initialized

Issue (2):
On PC the game stay forever on DLC progress bar when DLC is fully loaded. It occurs when one pack recursively processed another dependent pack from the manifest.

Changes (2):
Added missed lock.unlock() to avoid to stay on unlocked AZStd::unique_lock in the recursive call

## How was this PR tested?

DLC downloading was verified locally on PC and iPhone14 using env 'dev10' and enabled DLC via cvars:
```
mw_env dev10
mw_dlc_enabled true
```
Also the changes were verified with disabled DLC and with env 'qa' to be sure so that they don't impact current build (locally on PC and via iOS build 27847)

### CODE REVIEW

File: Code/Legacy/CryCommon/ILevelSystem.h [hunk 1]
```diff
@@ -58,6 +58,9 @@ struct ILevelSystem
 {
     virtual ~ILevelSystem() = default;
 
+#if defined (CARBONATED)
+    virtual void Activate() {}
```
Comment (Twolewis) [inline Code/Legacy/CryCommon/ILevelSystem.h]: Let's rename this to something besides Activate, as that is used/recognized mostly for Components & Entities.

**Review by Twolewis — CHANGES_REQUESTED** [review]

File: Code/Legacy/CrySystem/SystemInit.cpp [hunk 2]
```diff
@@ -1155,6 +1155,15 @@ bool CSystem::Init(const SSystemInitParams& startupParams)
     // Execute any deferred commands that uses the CVar commands that were just registered
     AZ::Interface<AZ::IConsole>::Get()->ExecuteDeferredConsoleCommands();
 
+#if defined (CARBONATED)
+    // Load the deffered level only after CrySystem has been fully initialized
+    AZ::TickBus::QueueFunction(
+       [this]()
+       { // Wait while RPI System is fully initialized
```
Comment (aefimov3) [inline Code/Legacy/CrySystem/SystemInit.cpp]: It is not clear how RPI system initialization is connected with CrySystem init, can you please clarify in the comment?

File: Code/Legacy/CryCommon/ILevelSystem.h (hunk 1 above)
Comment (vmednonogov) [reply Code/Legacy/CryCommon/ILevelSystem.h]: Renamed to LoadDefferedLevel()

File: Code/Legacy/CrySystem/SystemInit.cpp (hunk 2 above)
Comment (vmednonogov) [reply Code/Legacy/CrySystem/SystemInit.cpp]: The comment has been changed to be more understandable

**Review by aefimov3 — APPROVED** [review]

File: Code/Legacy/CryCommon/ILevelSystem.h [hunk 3]
```diff
@@ -58,6 +58,9 @@ struct ILevelSystem
 {
     virtual ~ILevelSystem() = default;
 
+#if defined (CARBONATED)
+    virtual void LoadDefferedLevel() {}
```
Comment (Twolewis) [inline Code/Legacy/CryCommon/ILevelSystem.h]: Spelling:  Deffered --> Deferred

**Review by Twolewis — APPROVED** [review]
Approved, but please fix spelling of deferred

File: Code/Legacy/CryCommon/ILevelSystem.h (hunk 3 above)
Comment (vmednonogov) [reply Code/Legacy/CryCommon/ILevelSystem.h]: fixed spelling

**Pull request merged**